src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java

```diff
@@ -0,0 +1,206 @@
+        );
+
+        CaseDetails exceptionRecordWithSearchFields =
+            exceptionRecord.toBuilder().data(exceptionRecordDataWithSearchFields).build();
+
+        CallbackRequest callbackRequest = CallbackRequest
+            .builder()
+            .eventId("createCase")
+            .caseDetails(exceptionRecordWithSearchFields)
+            .build();
+
+        CcdAuthenticator ccdAuthenticator = ccdAuthenticatorFactory.createForJurisdiction("BULKSCAN");
+
+        Response response = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(testUrl)
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Bulk Scan Orchestrator Functional test")
+            .header(AUTHORIZATION, ccdAuthenticator.getUserToken())
+            .header(CcdCallbackController.USER_ID, ccdAuthenticator.getUserDetails().getId())
+            .body(callbackRequest)
+            .when()
+            .post("/callback/create-case")
+            .andReturn();
+
+        return getCaseCcdId(response);
+    }
+
+    private String getCaseCcdId(Response response) throws java.io.IOException {
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        final AboutToStartOrSubmitCallbackResponse callbackResponse =
+            new ObjectMapper().readValue(response.getBody().asString(), AboutToStartOrSubmitCallbackResponse.class);
+        return (String)callbackResponse.getData().get("CASE_REFERENCE");
+    }
+
+    private Map<String, Object> exceptionRecordDataWithSearchFields(
+        CaseDetails exceptionRecord,
+        String searchCaseReferenceType
+    ) {
+        Map<String, Object> exceptionRecordData = new HashMap<>(exceptionRecord.getData());
+        exceptionRecordData.put("searchCaseReferenceType", searchCaseReferenceType);
+        return exceptionRecordData;
+    }
+
+    private CaseDetails createExceptionRecord(String resourceName) throws Exception {
+        UUID poBox = UUID.randomUUID();
+
+        envelopeMessager.sendMessageFromFile(resourceName, "0000000000000000", null, poBox, dmUrl);
+
+        await("Exception record is created")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(() -> lookUpExceptionRecord(poBox).isPresent());
+
+        return lookUpExceptionRecord(poBox).get();
+    }
+
+    private Optional<CaseDetails> lookUpExceptionRecord(UUID randomPoBox) {
+        List<CaseDetails> caseDetailsList = caseSearcher.search(
+            SampleData.JURSIDICTION,
+            SampleData.JURSIDICTION + "_" + CreateExceptionRecord.CASE_TYPE,
+            ImmutableMap.of(
+                "case.poBox", randomPoBox.toString()
+            )
+        );
+        return caseDetailsList.stream().findFirst();
+    }
+
+    private Boolean isCaseCreated(String caseCcdId, CaseDetails exceptionRecord, int expectedScannedDocsSize) {
```

Comment:
are we expecting more test cases with different number of documents ?

Reply:
This has been fixed